packages/gatsby-core-utils/src/fetch-remote-file.ts

```diff
@@ -105,11 +105,6 @@ function pollUntilComplete(
   buildId: string,
   cb: (err?: Error, result?: string) => void
 ): void {
-  if (!IS_WORKER) {
-    // We are not in a worker, so we shouldn't use the cache
-    return void cb()
-  }
-
```

Comment:
Looks like removing this return lead to broken tests:

https://app.circleci.com/pipelines/github/gatsbyjs/gatsby/77249/workflows/84d2bfc5-8a4b-4890-8eaf-8c253daaaa02/jobs/911184

```sh
 ● contentful extend node type › getBase64Image › keeps image format

    TypeError: Cannot read property 'then' of undefined

      76 |
      77 | function pollUntilComplete(cache, url, buildId, cb) {
    > 78 |   cache.get(cacheIdForWorkers(url)).then(entry => {
         |   ^
      79 |     if (!entry || entry.buildId !== buildId) {
      80 |       return void cb();
      81 |     }

      at pollUntilComplete (packages/gatsby-core-utils/dist/fetch-remote-file.js:78:3)
      at result (packages/gatsby-core-utils/dist/fetch-remote-file.js:114:5)
      at fetchFile (packages/gatsby-core-utils/dist/fetch-remote-file.js:113:24)
      at fetchRemoteFile (packages/gatsby-core-utils/dist/fetch-remote-file.js:69:24)
      at loadImage (packages/gatsby-source-contentful/src/gatsby-plugin-image.js:62:32)
      at getBase64Image (packages/gatsby-source-contentful/src/gatsby-plugin-image.js:72:19)
      at Object.<anonymous> (packages/gatsby-source-contentful/src/__tests__/gatsby-plugin-image.js:130:28)
```